src/tools/networkTools.cpp

```diff
@@ -211,39 +211,39 @@ std::map<std::string, std::string> getNetworkInterfaces() {
   return result;
 }
 
-
-std::string getBestPublicIp(bool ipv6) {
-  IpAddress bestPublicIp = IpAddress{"127.0.0.1","::1"};
+IpAddress getBestPublicIps(IpMode mode) {
+  IpAddress bestPublicIps = IpAddress{"127.0.0.1", "::1"};
   std::map<std::string, IpAddress> interfaces = getNetworkInterfacesIPv4Or6();
-
+  bool validMode4 = (mode == IpMode::ipv4 || mode == IpMode::all);  // Mode is valid to support ipv4
+  bool validMode6 = (mode == IpMode::ipv6 || mode == IpMode::all);  // Mode is valid to support ipv6
 #ifndef _WIN32
   const char* const prioritizedNames[] = { "eth0", "eth1", "wlan0", "wlan1", "en0", "en1" };
-  for(auto name: prioritizedNames) {
-    auto it=interfaces.find(name);
-    if(it != interfaces.end() && !(ipv6 && (*it).second.addr6.empty())) {
-      bestPublicIp = (*it).second;
-      break;
-    }
+  for(const auto& name: prioritizedNames) {
+    const auto it = interfaces.find(name);
+    if(it == interfaces.end()) continue;
+    const IpAddress& interfaceIps = it->second;
+    if(!bestPublicIps.addr.empty() && validMode4 && !interfaceIps.addr.empty()) 
+      bestPublicIps.addr = interfaceIps.addr;
```

Comment:
Shouldn't the condition be `if ( bestPublicIps.addr.empty() ... )` so that `bestPublicIps.addr` is set only once?

The same bug is present for the rest of the similar `if`s in this function.

Reply:
You are totally correct. Thanks for catching this...